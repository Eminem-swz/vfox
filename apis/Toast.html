<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Toast 轻提示 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.5312ab12.css">
    <link rel="modulepreload" href="/vfox/assets/Home.89ab9d66.js">
    <link rel="modulepreload" href="/vfox/assets/app.76ef83d6.js">
    <link rel="modulepreload" href="/vfox/assets/apis_Toast.md.910bc1f7.lean.js">
    
    <meta name="twitter:title" content="Toast 轻提示 | Vfox for Vue3">
  <meta property="og:title" content="Toast 轻提示 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">通知类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/apis/Toast">Toast</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#showtoast-object">showToast(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#hidetoast-object">hideToast([object])</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object-1">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-1">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#showloading-object">showLoading(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object-2">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-2">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#hideloading-object">hideLoading([object])</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object-3">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-3">Usage</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Notify">Notify</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">预览类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Preview">Preview</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">模态框类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Dialog">Dialog</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">抽屉类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Calendar">Calendar</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">气泡类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopMenu">PopMenu</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div class="demo"><iframe style="display:none;" src="https://godxiaoji.github.io/vfox/demo/#/" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="toast-轻提示" tabindex="-1">Toast 轻提示 <a class="header-anchor" href="#toast-轻提示" aria-hidden="true">#</a></h1><p>注：目前仅支持单例模式。</p><h2 id="showtoast-object" tabindex="-1">showToast(object) <a class="header-anchor" href="#showtoast-object" aria-hidden="true">#</a></h2><p>显示消息提示框。</p><h3 id="object" tabindex="-1">object <a class="header-anchor" href="#object" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>是</td><td>提示的内容</td></tr><tr><td>type</td><td>string</td><td>&#39;default&#39;</td><td>否</td><td>类型，可选 &#39;success&#39;, &#39;loading&#39;, &#39;fail&#39;</td></tr><tr><td>icon</td><td>string</td><td></td><td>否</td><td>图标，使用 <a href="./../components/Icon.html">Icon</a> 组件，优先级高于 <code>type</code></td></tr><tr><td>duration</td><td>number</td><td>1500</td><td>否</td><td>提示的延迟关闭时间</td></tr><tr><td>mask</td><td>boolean</td><td>false</td><td>否</td><td>是否显示透明蒙层，防止触摸穿透</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>接口调用结束的回调函数（调用成功、失败都会执行）</td></tr></tbody></table><h4 id="object-type-的合法值" tabindex="-1">object.type 的合法值 <a class="header-anchor" href="#object-type-的合法值" aria-hidden="true">#</a></h4><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>default</td><td>不显示图标，此时 title 文本最多可显示两行</td></tr><tr><td>success</td><td>显示成功图标，此时 title 文本最多显示 7 个汉字长度</td></tr><tr><td>loading</td><td>显示加载图标，此时 title 文本最多显示 7 个汉字长度</td></tr><tr><td>fail</td><td>显示错误图标，此时 title 文本最多显示 7 个汉字长度</td></tr></tbody></table><h3 id="usage" tabindex="-1">Usage <a class="header-anchor" href="#usage" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">showToast</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  <span class="token literal-property property">title</span><span class="token operator">:</span> <span class="token string">&#39;成功&#39;</span><span class="token punctuation">,</span>
  <span class="token literal-property property">type</span><span class="token operator">:</span> <span class="token string">&#39;success&#39;</span><span class="token punctuation">,</span>
  <span class="token literal-property property">duration</span><span class="token operator">:</span> <span class="token number">2000</span>
<span class="token punctuation">}</span><span class="token punctuation">)</span>
</code></pre></div><h2 id="hidetoast-object" tabindex="-1">hideToast([object]) <a class="header-anchor" href="#hidetoast-object" aria-hidden="true">#</a></h2><p>隐藏消息提示框。</p><h3 id="object-1" tabindex="-1">object <a class="header-anchor" href="#object-1" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败的回调函数</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>接口调用结束的回调函数（调用成功、失败都会执行）</td></tr></tbody></table><h3 id="usage-1" tabindex="-1">Usage <a class="header-anchor" href="#usage-1" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">hideToast</span><span class="token punctuation">(</span><span class="token punctuation">)</span>
</code></pre></div><h2 id="showloading-object" tabindex="-1">showLoading(object) <a class="header-anchor" href="#showloading-object" aria-hidden="true">#</a></h2><p>显示 loading 提示框。需主动调用 <a href="./Toast.html#toast-hideloading-object">hideLoading</a> 才能关闭提示框。</p><h3 id="object-2" tabindex="-1">object <a class="header-anchor" href="#object-2" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>是</td><td>提示的内容</td></tr><tr><td>mask</td><td>boolean</td><td>false</td><td>否</td><td>是否显示透明蒙层，防止触摸穿透</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败的回调函数</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>接口调用结束的回调函数（调用成功、失败都会执行）</td></tr></tbody></table><h3 id="usage-2" tabindex="-1">Usage <a class="header-anchor" href="#usage-2" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">showLoading</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  <span class="token literal-property property">title</span><span class="token operator">:</span> <span class="token string">&#39;加载中&#39;</span><span class="token punctuation">,</span>
<span class="token punctuation">}</span><span class="token punctuation">)</span>
</code></pre></div><h2 id="hideloading-object" tabindex="-1">hideLoading([object]) <a class="header-anchor" href="#hideloading-object" aria-hidden="true">#</a></h2><p>隐藏 loading 提示框。</p><h3 id="object-3" tabindex="-1">object <a class="header-anchor" href="#object-3" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败的回调函数</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>接口调用结束的回调函数（调用成功、失败都会执行）</td></tr></tbody></table><h3 id="usage-3" tabindex="-1">Usage <a class="header-anchor" href="#usage-3" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">hideLoading</span><span class="token punctuation">(</span><span class="token punctuation">)</span>
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><!----></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/apis/Notify" data-v-38ede35f><span class="text" data-v-38ede35f>Notify</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_actionsheet.md\":\"48a7a313\",\"apis_calendar.md\":\"0061176b\",\"apis_cascader.md\":\"852ec8d5\",\"apis_datepicker.md\":\"a94b760f\",\"apis_dialog.md\":\"b6619b0a\",\"apis_notify.md\":\"8368bad0\",\"apis_picker.md\":\"71e6a40b\",\"apis_popdialog.md\":\"65815c20\",\"apis_popmenu.md\":\"867b427f\",\"apis_popover.md\":\"b894f868\",\"apis_preview.md\":\"4ddeeacd\",\"apis_toast.md\":\"910bc1f7\",\"apis_index.md\":\"d3988204\",\"components_actionsheet.md\":\"4fa4926b\",\"components_activityindicator.md\":\"f50bd7cd\",\"components_avatar.md\":\"7c7325b7\",\"components_backtop.md\":\"a3c5683d\",\"components_badge.md\":\"e3cabc65\",\"components_button.md\":\"97865831\",\"components_calendar.md\":\"849d90a8\",\"components_cascader.md\":\"0dcae51b\",\"components_cell.md\":\"1bb665d1\",\"components_checkbox.md\":\"56f240a9\",\"components_circleprogress.md\":\"7f22e8e8\",\"components_collapse.md\":\"bcd5f8a0\",\"components_copy.md\":\"fc2025e7\",\"components_countdown.md\":\"d002712c\",\"components_countup.md\":\"864ed525\",\"components_datepicker.md\":\"b32ea26b\",\"components_dialog.md\":\"f06ed975\",\"components_divider.md\":\"90ee9e6a\",\"components_drawer.md\":\"fafe0d69\",\"components_dropdown.md\":\"55c785b1\",\"components_empty.md\":\"43cfd231\",\"components_fixed.md\":\"6c67067b\",\"components_flatlist.md\":\"ae091029\",\"components_form.md\":\"32619e35\",\"components_group.md\":\"657dfd05\",\"components_icon.md\":\"d682f3ec\",\"components_image.md\":\"e5d33d43\",\"components_imagepreview.md\":\"cfbadc50\",\"components_imageuploader.md\":\"5fb5ff79\",\"components_indexview.md\":\"02b04833\",\"components_input.md\":\"5a4ceda9\",\"components_layout.md\":\"f47f33f3\",\"components_loadmore.md\":\"058fb73f\",\"components_modal.md\":\"794a5187\",\"components_navbar.md\":\"dd7c1084\",\"components_noticebar.md\":\"6c12bacd\",\"components_notify.md\":\"f1f6a490\",\"components_numberkeyboard.md\":\"cf0ffdc1\",\"components_order.md\":\"834f0076\",\"components_pagination.md\":\"954fbc6b\",\"components_picker.md\":\"baf018c9\",\"components_popdialog.md\":\"5fceed17\",\"components_popmenu.md\":\"1ac20a58\",\"components_popover.md\":\"55eedbe0\",\"components_price.md\":\"b86607a3\",\"components_progress.md\":\"5da1c322\",\"components_radio.md\":\"83281d83\",\"components_range.md\":\"e2c4e51d\",\"components_rate.md\":\"db20c4ab\",\"components_result.md\":\"5e57bd29\",\"components_scrolltab.md\":\"2ac91b36\",\"components_scrollview.md\":\"688d9a1b\",\"components_searchbar.md\":\"32c26d22\",\"components_sidetab.md\":\"e0ce476b\",\"components_skeleton.md\":\"7f803023\",\"components_slider.md\":\"eb42867f\",\"components_stepper.md\":\"2658f905\",\"components_steps.md\":\"efbe1a67\",\"components_sticky.md\":\"a467939e\",\"components_stopwatch.md\":\"d37667aa\",\"components_swipecell.md\":\"a427f63e\",\"components_swiper.md\":\"1bef4cc1\",\"components_switch.md\":\"3698d99e\",\"components_tab.md\":\"2629c4b2\",\"components_tabbar.md\":\"e1ff7884\",\"components_tabview.md\":\"ad2c38e9\",\"components_tag.md\":\"fef6bbf0\",\"components_timeago.md\":\"26644300\",\"components_timeline.md\":\"1ce28894\",\"components_toast.md\":\"73a79f4d\",\"components_index.md\":\"9b8bbcc9\",\"design_color.md\":\"48eefa92\",\"design_index.md\":\"b0027359\",\"guide_dark.md\":\"b68caf8f\",\"guide_import.md\":\"00d662ab\",\"guide_locale.md\":\"50f29438\",\"guide_theme.md\":\"95ebb344\",\"index.md\":\"514c1bf9\"}")</script>
    <script type="module" async src="/vfox/assets/app.76ef83d6.js"></script>
    
  </body>
</html>