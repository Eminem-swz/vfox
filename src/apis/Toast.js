import {
  isObject,
  isString,
  inArray,
  isUndefined,
  noop,
  isFunction
} from '../helpers/util'

function render(options) {
  const html = [
    `<div class="toast ` +
      (options.image || (options.icon && options.icon !== 'none')
        ? 'has-icon'
        : '') +
      `">`,
    options.mask ? `<div class="toast-mask"></div>` : '',
    `<div class="toast-box">`,
    options.image
      ? `<img class="toast-icon" src="${options.image}" />`
      : options.icon === 'success'
      ? `<i class="toast-icon">
        <svg class="icon" width="48px" height="48.00px" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg"><path d="M887.904744 298.20852c-12.863647-12.063755-33.151673-11.487488-45.215428 1.408843L415.935493 753.983819 182.815858 524.287381c-12.607338-12.416396-32.8644-12.287381-45.280796 0.319957-12.416396 12.576374-12.256417 32.8644 0.352641 45.248112l256.479935 252.671415c0.096331 0.096331 0.223626 0.127295 0.319957 0.223626s0.127295 0.223626 0.223626 0.319957c2.016073 1.919742 4.448434 3.008628 6.784464 4.288456 1.152533 0.672598 2.143368 1.663432 3.359548 2.143368 3.775837 1.47249 7.775299 2.239699 11.743798 2.239699 4.192125 0 8.384249-0.832576 12.287381-2.496009 1.312512-0.543583 2.33603-1.663432 3.552211-2.368714 2.399677-1.408843 4.895686-2.59234 6.944443-4.67206 0.096331-0.096331 0.127295-0.25631 0.223626-0.352641 0.063647-0.096331 0.192662-0.127295 0.287273-0.223626L889.277463 343.420508C901.439269 330.591265 900.768391 310.335923 887.904744 298.20852z" /></svg>
      </i>`
      : options.icon === 'loading'
      ? `<i class="toast-icon loading-icon"><svg class="icon" width="48px" height="48.00px" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg"><path fill="#ffffff" d="M515.698303 969.127499c-97.187972 0-191.279691-31.134554-270.406182-90.479422-96.67193-72.245926-159.45708-178.206619-176.658492-297.928439s13.245087-238.9276 85.663027-335.59953C304.120947 45.239711 588.288258 4.644381 787.99664 154.124643c83.770872 62.78515 143.459768 153.092558 168.2298 254.580884 4.300353 17.373425-6.364522 34.918864-23.737947 39.047203-17.373425 4.128339-34.918864-6.364522-39.047203-23.737947-21.157736-86.867126-72.417941-164.44549-144.147825-218.285906C578.139425 77.750378 334.395431 112.669242 206.244919 283.823282c-62.097094 82.910801-88.243239 185.087183-73.450025 287.607593s68.461616 193.34386 151.372417 255.440954c171.326054 128.322526 414.898035 93.403662 543.220561-77.922392 33.542752-44.895683 56.592642-95.123803 68.289602-149.308248 3.78431-17.373425 21.157736-28.554342 38.359147-24.770032 17.373425 3.78431 28.554342 20.985721 24.770032 38.359147-13.761129 63.473207-40.59533 122.130018-79.814547 174.422308-72.417941 96.67193-178.378633 159.45708-298.100454 176.658492C559.217873 967.579372 537.372081 969.127499 515.698303 969.127499z" /></svg>
        </i>`
      : '',
    `<div class="toast-text">${options.title}</div>`,
    `</div>`,
    `</div>`
  ].join('')

  const $wrapper = document.createElement('div')

  $wrapper.innerHTML = html

  return $wrapper.firstChild
}

function getParamValue(object, key, type, defaultValue) {
  const param = object[key]

  if (!isUndefined(param)) {
    if (typeof param === type) {
      return param
    } else {
      throw new TypeError(
        `Invalid param: "object.${key}" must be a ${type} type.`
      )
    }
  } else {
    return defaultValue
  }
}

const ICON_NAMES = ['success', 'loading', 'none']

var timer = null
var $el = null

export function show(object) {
  if (!isObject(object)) {
    object = {}
  }

  const options = {}
  const complete = isFunction(object.complete) ? object.complete : noop

  try {
    if (!isString(object.title)) {
      throw new TypeError(
        'Invalid param: "object.title" must be a string type.'
      )
    }
    options.title = object.title

    options.image = getParamValue(object, 'image', 'string', null)
    options.mask = getParamValue(object, 'mask', 'boolean', false)
    options.duration = getParamValue(object, 'duration', 'number', 1500)

    if (!isUndefined(object.icon)) {
      if (inArray(object.icon, ICON_NAMES)) {
        options.icon = object.icon
      } else {
        const iconStr = ICON_NAMES.join(`","`)
        throw new TypeError(
          `Invalid param: "object.icon" must be "${iconStr}".`
        )
      }
    } else {
      options.icon = null
    }

    _hide()

    $el = render(options)
    $el._options = options
    document.body.appendChild($el)

    // 定时隐藏
    if (options.duration > 0) {
      timer = setTimeout(() => {
        _hide()
      }, options.duration)
    }

    if (isFunction(object.success)) {
      object.success({})
    }
    complete({})
  } catch (e) {
    const res = {
      errMsg: e.message
    }

    if (isFunction(object.fail)) {
      object.fail(res)
      complete(res)
    } else {
      complete(res)
      throw e
    }
  }
}

function _hide() {
  clearTimeout(timer)

  if ($el) {
    document.body.removeChild($el)
    $el = null
  }
}

export function hide(object) {
  clearTimeout(timer)

  if (!isObject(object)) {
    object = {}
  }

  const complete = isFunction(object.complete) ? object.complete : noop

  try {
    if ($el) {
      document.body.removeChild($el)
      $el = null
    }

    if (isFunction(object.success)) {
      object.success({})
    }
    complete({})
  } catch (e) {
    const res = {
      errMsg: e.message
    }

    if (isFunction(object.fail)) {
      object.fail(res)
      complete(res)
    } else {
      complete(res)
      throw e
    }
  }
}
