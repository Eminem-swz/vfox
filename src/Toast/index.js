import Toast from './Toast.vue'
import { showToast, showLoading, hideLoading, hideToast } from '../apis/Toast'

Toast.install = function(Vue) {
  Vue.component(Toast.name, Toast)

  // 添加api
  Vue.prototype.$showToast = showToast
  Vue.prototype.$hideToast = hideToast
  Vue.prototype.$showLoading = showLoading
  Vue.prototype.$hideLoading = hideLoading
}

Toast.showToast = showToast
Toast.hideToast = hideToast
Toast.showLoading = showLoading
Toast.hideLoading = hideLoading

export default Toast
