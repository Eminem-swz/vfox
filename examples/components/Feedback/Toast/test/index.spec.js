/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Toast from '@/Toast'

describe('Toast', () => {
  test('should render default correctly', () => {
    const wrapper = mount(Toast, {
      props: {
        title: '提示文本'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop type="success" correctly', () => {
    const wrapper = mount(Toast, {
      props: { title: '成功文本', type: 'success' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop type="fail" correctly', () => {
    const wrapper = mount(Toast, {
      props: { title: '失败文本', type: 'fail' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop type="loading" correctly', () => {
    const wrapper = mount(Toast, {
      props: { title: '加载文本', type: 'loading' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
