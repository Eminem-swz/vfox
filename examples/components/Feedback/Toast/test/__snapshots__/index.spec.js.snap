// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render default correctly 1`] = `
<div class="fx-toast fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-toast_box">
    <!---->
    <div class="fx-toast_text">
      提示文本
    </div>
  </div>
</div>
`;

exports[`Toast should render prop type="fail" correctly 1`] = `
<div class="fx-toast fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-toast_box has--icon"><svg class="fx-icon fx-toast_icon">
      <use xlink:href="#icon-CloseCircleOutlined"></use>
    </svg>
    <div class="fx-toast_text">
      失败文本
    </div>
  </div>
</div>
`;

exports[`Toast should render prop type="loading" correctly 1`] = `
<div class="fx-toast fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-toast_box has--icon"><svg class="fx-icon fx-toast_icon spin">
      <use xlink:href="#icon-LoadingOutlined"></use>
    </svg>
    <div class="fx-toast_text">
      加载文本
    </div>
  </div>
</div>
`;

exports[`Toast should render prop type="success" correctly 1`] = `
<div class="fx-toast fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-toast_box has--icon"><svg class="fx-icon fx-toast_icon">
      <use xlink:href="#icon-CheckCircleOutlined"></use>
    </svg>
    <div class="fx-toast_text">
      成功文本
    </div>
  </div>
</div>
`;
